Sources/SpelledPitch/PitchSpeller/Wetherfield/WeightedGraphSchemeProtocol.swift

```diff
@@ -21,25 +23,84 @@ extension WeightedGraphSchemeProtocol {
         H: WeightedGraphSchemeProtocol,
         H.Weight == Weight
     {
-        return H.init { self.weight(Edge(f($0.a),f($0.b))) }
+        return H { self.weight(from: f($0.a), to: f($0.b)) }
     }
-}
 
-extension WeightedGraphSchemeProtocol {
     @inlinable
     func unweighted <H> () -> H where
         H: UnweightedGraphSchemeProtocol,
         H.Edge == Edge
     {
-        return H.init { self.weight($0) != nil }
+        return H { self.weight($0) != nil }
+    }
+}
+
+extension WeightedGraphSchemeProtocol where Self: UndirectedGraphSchemeProtocol, Weight: Numeric {
+    
+    static func * (lhs: Self, rhs: Self) -> Self {
+        return Self { edge in
+            guard let lweight = lhs.weight(edge), let rweight = rhs.weight(edge) else { return nil }
+            return lweight * rweight
+        }
+    }
+    
+    static func + (lhs: Self, rhs: Self) -> Self {
+        return Self { edge in
+            if let lweight = lhs.weight(edge) {
+                if let rweight = rhs.weight(edge) { return lweight + rweight }
+                return lweight
+            } else {
+                guard let rweight = rhs.weight(edge) else { return nil }
+                return rweight
+            }
+        }
     }
 }
 
-extension WeightedGraphProtocol {
-    func weightScheme <G> (_ f: @escaping (G.Node) -> Node) -> WeightCarrying<G> where
-        G: WeightedGraphProtocol,
-        G.Weight == Weight
+extension WeightedGraphSchemeProtocol where Self: DirectedGraphSchemeProtocol, Weight: Numeric {
+    
+    static func * <Scheme> (lhs: Self, rhs: Scheme) -> Self where
+    Scheme: WeightedGraphSchemeProtocol,
+    Scheme.Node == Node,
+    Scheme.Weight == Weight
     {
-        return WeightCarrying.build(from: self).pullback(f)
+        return Self { edge in
+            guard
```

Comment:
Here, I like to have the `guard` and the `else` aligned:

```Swift
guard 
    let lweight = lhs.weight(edge),
    let rweight = rhs.weight(from: edge.a, to: edge.b)
else {
     return nil
}
```

Reply:
However, this could be shortened to:

```Swift
return lhs.weight(edge) ?? rhs.weight(from edge.a, to edge.b)
```

Reply:
Cool!

Reply:
hold on, but does this capture the logic of when they are both non-nil ?